test/multiple-user-principal-types.test.js

```diff
@@ -0,0 +1,485 @@
+      var options = {
+        email: 'foo@bar.com',
+        redirect: 'http://foobar.com/reset-password',
+      };
+
+      it('should create a temp accessToken to allow a user to change password', function(done) {
+        OneUser.resetPassword({email: options.email})
+        .catch(function(err) {
+          assert(!err);
+        });
+
+        OneUser.once('resetPasswordRequest', function(info) {
+          assertGoodToken(info.accessToken, userFromOneModel);
+          info.accessToken.user.getAsync()
+          .then(function(user) {
+            assert.equal(user.id, userFromOneModel.id);
+            assert.equal(user.email, userFromOneModel.email);
+            done();
+          })
+          .catch(done);
+        });
+      });
+    });
+  });
+
+  describe('AccessToken (session) invalidation when changing email', function() {
+    var anotherUserFromOneModel;
+
+    it('should only impact related user', function(done) {
+      Promise.all([
+        OneUser.create({email: 'original@example.com', password: 'bar'}),
+      ])
+      .spread(function(u1, u2, u3) {
+        anotherUserFromOneModel = u1;
+        return Promise.all([
+          OneUser.login({email: 'original@example.com', password: 'bar'}),
+          OneUser.login(commonCredentials),
+          AnotherUser.login(commonCredentials),
+        ]);
+      })
+      .then(function() {
+        return anotherUserFromOneModel.updateAttributes({email: 'updated@example.com'});
+      })
+      .then(function() {
+        return Promise.all([
+          AccessToken.find({where: {userId: anotherUserFromOneModel.id,
+            principalType: OneUser.modelName}}),
+          AccessToken.find({where: {userId: userFromOneModel.id,
+            principalType: OneUser.modelName}}),
+          AccessToken.find({where: {userId: userFromAnotherModel.id,
+            principalType: AnotherUser.modelName}}),
+        ]);
+      })
+      .spread(function(anotherUserFromOneModelTokens, userFromOneModelTokens,
+      userFromAnotherModelTokens) {
+        expect(anotherUserFromOneModelTokens.length).to.equal(0);
+        expect(userFromOneModelTokens.length).to.be.above(0);
+        expect(userFromAnotherModelTokens.length).to.be.above(0);
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('role model:', function() {
+    this.timeout(10000);
+
+    var Role, RoleMapping, ACL, user, userRole;
+
+    beforeEach(function() {
+      ACL = app.registry.getModel('ACL');
+      app.model(ACL, {dataSource: 'db'});
+
+      Role = app.registry.getModel('Role');
+      app.model(Role, {dataSource: 'db'});
+
+      RoleMapping = app.registry.getModel('RoleMapping');
+      app.model(RoleMapping, {dataSource: 'db'});
+    });
+
+    describe('role/user relations:', function() {
+      beforeEach(function(done) {
+        OneUser.create({name: 'Eric', email: 'x@y.com', password: 'foobar'})
+        .then(function(u) {
```

Comment:
i hope i got you right, i added one indent on **every** first `then` following a single lined calls